ieeeuoft/hackathon-template
# Hardware signout site time restrictions

## Overview

- Added date restrictions to frontend and backend
- participants can't submit orders if not within hardware sign out period
- an info alert message will appear on the frontend if outside of time period
- add to cart and submit order buttons are disabled

<img width="932" alt="image" src="https://user-images.githubusercontent.com/20618460/177435575-a7726331-2f50-4420-a98b-64252f10215a.png">

## Unit Tests Created

- tests for submit order api with date restrictions
- tests for info alert message displaying properly
- tests for disabled buttons


## Steps to QA

- in settings/__init__.py change `HARDWARE_SIGN_OUT_START_DATE` and `HARDWARE_SIGN_OUT_END_DATE` to see restrictions take effect
- in constants.js change `hardwareSignOutStartDate` and `hardwareSignOutEndDate` to see alert boxes appear
